sonar-lits-plugin/src/main/java/com/sonarsource/lits/IssuesChecker.java

```diff
@@ -166,7 +166,7 @@ void save() {
       exception = MessageException.of(message);
     }
     if (!missingResources.isEmpty()) {
-      String message = "Missing resources: " + Joiner.on(", ").join(missingResources);
+      String message = "Files listed in Expected directory were not analyzed: " + Joiner.on(", ").join(missingResources);
```

Comment:
maybe "The following files which are listed in Expected directory were not analyzed:"?

and do we know why? what does `missingResources` contain? are e.g. they files from a CSPROJ that are not found on disk?

Reply:
I wanted the message to be as short as possible, because the error line will be probably very long (all filenames are on the same line).

This is how the IT failed, no more lines were written after this:

> INFO: ------------------------------------------------------------------------
> INFO: Total time: 8.315s
> INFO: Final Memory: 9M/37M
> INFO: ------------------------------------------------------------------------
> ERROR: Error during SonarQube Scanner execution
> ERROR: Missing resources: InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/Dictionary.Write.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/SqlInjection.WebControls.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/HTTPHeaderResponse.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/Source.WinForms.RichTextBox.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/SqlInjection.EntityFramework.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/Source.UnvalidatedRequestValuesWrapper.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/SqlInjection.Linq.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/ReflectedXss.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/Source.AspNetMvcController.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/SqlInjection.SqlServerCe.cs, InternalBasic.CSharp.NetCore3:InternalBasic.CSharp.NetFramework/SqlInjection.MySql.cs, 
> ERROR: 
> The SonarQube Scanner did not complete successfully
> 14:14:10.025  Post-processing failed. Exit code: 1

This happens when:
- File exist in directory structure of analyzed project
- File contains `// Noncompliant` comments
- Expected line number is written to the `Expected` directory into the `json` file.
- File is not analyzed.

In .NET scenario, this can happen when:
- We forget to include file into the project
or
- We play with [ExpectedIssues.java / ensureCorrectCSharpFilesAreSelected](https://github.com/SonarSource/sonar-security/blob/d61453a22c03e558f7b6d781c3462f3853f4e0a6/its/src/test/java/com/sonar/security/ExpectedIssues.java#L90)

Reply:
+1 for keeping the message as it is

However, I think you can add a comment with

```
// This could happen when:
// - File exist in directory structure of analyzed project
// - File contains '// Noncompliant' comments
// - Expected line number is written to the Expected directory into the json file.
// - File is not analyzed.
```

Lot's of people work with sonar-security and many don't know how the LITS framework works, so such a comment will be a time-saver.

And there's a low chance that we will refactor a lot the way we parse these tests to end up with a stale comment.

- The people that know how it works will figure things out from the error message.
- The people that don't know how it works will bless the dev that left this comment :)

Reply:
Problem is that the error terminates the analysis in the middle of nowhere. It took me 3 days and I had to bother 2 other devs just to read the big picture of the logs properly to find out that that it actually comes from LITS. We were searching in the scanner and in the plugin first, unable to find where the message comes from.

Having a better message would solve the original issue immediately. 

I can add the comment, but I really want to change the error message. Because it's useless as it is now. It's talking about internal state of this (not very well known) plugin instead of telling what actually went wrong and why.